packages/react-components/react-table/src/utils/columnResizeUtils.ts

```diff
@@ -0,0 +1,204 @@
+ *  - It returns old state if no changes in the state have been made (so that react doesn't call effects)
+ * @param columns
+ * @param state
+ * @returns
+ */
+
+export function columnDefinitionsToState<T>(
+  columns: TableColumnDefinition<T>[],
+  state: ColumnWidthState[] = [],
+  columnSizingOptions: TableColumnSizingOptions = {},
+): ColumnWidthState[] {
+  let updated = false;
+
+  let updatedState = columns.map(column => {
+    const { columnId } = column;
+    const existingColumnState = state.find(col => col.columnId === column.columnId);
+
+    if (existingColumnState) {
+      const newIdealWidth = columnSizingOptions[column.columnId]?.idealWidth;
+      const newMinWidth = columnSizingOptions[column.columnId]?.minWidth;
+      const newPadding = columnSizingOptions[column.columnId]?.padding;
+      if (
+        (newIdealWidth && newIdealWidth !== existingColumnState.idealWidth) ||
+        (newMinWidth && newMinWidth !== existingColumnState.minWidth) ||
+        (newPadding && newPadding !== existingColumnState.padding)
+      ) {
+        updated = true;
+        return {
+          ...existingColumnState,
+          idealWidth: columnSizingOptions[column.columnId]?.idealWidth ?? existingColumnState.idealWidth,
+          width: columnSizingOptions[column.columnId]?.idealWidth ?? existingColumnState.idealWidth,
+          minWidth: columnSizingOptions[column.columnId]?.minWidth ?? existingColumnState.minWidth,
+          padding: columnSizingOptions[column.columnId]?.padding ?? existingColumnState.padding,
+        };
+      }
+
+      return existingColumnState;
+    }
+
+    updated = true;
+    return {
+      columnId,
+      width:
+        columnSizingOptions[column.columnId]?.defaultWidth ??
+        columnSizingOptions[column.columnId]?.idealWidth ??
+        DEFAULT_WIDTH,
+      minWidth: columnSizingOptions[column.columnId]?.minWidth ?? DEFAULT_MIN_WIDTH,
+      idealWidth:
+        columnSizingOptions[column.columnId]?.defaultWidth ??
+        columnSizingOptions[column.columnId]?.idealWidth ??
+        DEFAULT_WIDTH,
+      padding: columnSizingOptions[column.columnId]?.padding ?? 16,
+    };
+  });
+
+  if (updatedState.length !== state.length) {
+    // Adding or removing columns, set all columns which have a different idealWidth than width to width = idealWidth,
+    // so that the adjustColumnWidthsToFitContainer can successfully expand the last column,
+    // since the column which was last before is not necessarily last now.
+    if (updatedState.length > state.length) {
+      updatedState = updatedState.map(s => {
+        if (s.idealWidth !== s.width) {
+          s.width = s.idealWidth;
+        }
+        return s;
+      });
+    }
+
+    return updatedState;
+  }
+
+  const a1 = state.map(({ columnId }) => columnId);
+  const a2 = updatedState.map(({ columnId }) => columnId);
+
+  if (!a1.every((v, i) => v === a2[i])) {
+    updated = true;
+  }
```

Comment:
if you're going to do a call of `every` anyway then there is no need for the `updated` flag to be used earlier since you can just extend the `every` check for object references too